tools/execsnoop.py

```diff
@@ -24,14 +24,35 @@
 import argparse
 import re
 import time
+import pwd
 from collections import defaultdict
 from time import strftime
 
+
+def parse_uid(user):
+    try:
+        result = int(user)
+    except ValueError:
+        try:
+            user_info = pwd.getpwnam(user)
+        except KeyError:
+            raise argparse.ArgumentTypeError(
+                "{0!r} is not valid UID or user entry".format(user))
+        else:
+            return user_info.pw_uid
+    else:
+        # Maybe validate if UID < 0 ?
+        return result
+
+
 # arguments
 examples = """examples:
     ./execsnoop           # trace all exec() syscalls
     ./execsnoop -x        # include failed exec()s
     ./execsnoop -T        # include time (HH:MM:SS)
+    ./execsnoop -U        # include UID
+    ./execsnoop -u 1000   # only trace UID 1000
+    ./execsnoop -u user   # get user UID and trace only them
```

Comment:
You do not need this one `... -u user`.

Reply:
I just wanted to emphasise that both: UID and username could be specified as option there.

Reply:
I replaced `user` with `root`, now it should be obvious that this is user login name.